File: JAVA/SRobotics/src/sRobotics/core/Context.java
package sRobotics.core;

import sRobotics.core.state.ActivationStateType;

public class Context
{
    public Robot currentRobot;

    public void ChangeRobot(Robot robot)
    {
        currentRobot = robot;
    }

    public void TurnOn()
    {
        currentRobot.ChangeActivationState(ActivationStateType.on);
    }

    public void TurnOff()
    {
        currentRobot.ChangeActivationState(ActivationStateType.off);
    }
}

File: C#/SRobotics/Core/Context.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core
{
    public class Context
    {
        public Robot CurrentRobot { get; private set; }

        public void ChangeRobot(Robot robot)
        {
            CurrentRobot = robot;
        }

        public void TurnOn()
        {
            CurrentRobot.ChangeActivationState(State.ActivationStateType.On);
        }

        public void TurnOff()
        {
            CurrentRobot.ChangeActivationState(State.ActivationStateType.Off);
        }
    }
}
